flatlaf-core/src/main/java/com/formdev/flatlaf/UIDefaultsLoader.java

```diff
@@ -351,14 +351,15 @@ else if( value.startsWith( "\"" ) && value.endsWith( "\"" ) ) {
 				valueType = ValueType.STRING;
 			else if( key.endsWith( "Color" ) ||
 				(key.endsWith( "ground" ) &&
-				 (key.endsWith( ".background" ) || key.endsWith( "Background" ) ||
-				  key.endsWith( ".foreground" ) || key.endsWith( "Foreground" ))) )
+				 (key.endsWith( ".background" ) || key.endsWith( "Background" ) || key.equals( "background" ) ||
+				  key.endsWith( ".foreground" ) || key.endsWith( "Foreground" ) || key.equals( "foreground" ))) )
```

Comment:
Wouldn't `endsWithIgnorecase( "foreground"/"background")` suffice and also be faster?
Same for the other instances where you have this pattern?

Reply:
Method `endsWithIgnorecase()` does not exist 😄 

And following line avoids that this is executed in most cases:

https://github.com/JFormDesigner/FlatLaf/blob/82192bef913254755f43752a5e74c7f4a65b3957/flatlaf-core/src/main/java/com/formdev/flatlaf/UIDefaultsLoader.java#L353

`equals( "background" )` is very fast because it first compares the string lengths.
`startsWith()` and `endsWith()` are also very fast because they return immediately when a character is not equal.

Reply:
That's absolutely true :D